reactivesocket/src/test/scala/akka/stream/alpakka/reactivesocket/scaladsl/ReactiveSocketSpec.scala

```diff
@@ -0,0 +1,94 @@
+package akka.stream.alpakka.reactivesocket.scaladsl
+
+import akka.stream.scaladsl._
+import io.reactivesocket.util.PayloadImpl
+import org.scalatest.WordSpec
+
+import scala.concurrent.duration._
+
+class ReactiveSocketSpec extends WordSpec {
+
+  "reactive socket API" should {
+
+    "support server and client" in {
+
+      import io.reactivesocket.transport.tcp.client.TcpTransportClient
+      import io.reactivesocket.transport.tcp.server.TcpTransportServer
+
+      val binding = ReactiveSocketServer(TcpTransportServer.create())
+          .toMat(Sink.foreach {
+            _.flow.runWith(Source.repeat(new PayloadImpl("Pong")), Sink.foreach(println))
```

Comment:
The alternative is to encode it in the elements, different types of `IncomingConnection`, but I couldn't see a way to describe in the types that for example request-response should only return one element.

I also like the idea of that client and server interfaces are the same.